### PR TITLE
Remove support for bulk read/write of attributes

### PR DESCRIPTION
**Pull request by pcercuei**
Remove `iio_{device,channel,device_debug,device_buffer}_attr_{read,write}_all` functions for a big code cleanup, and because they are getting in the way of the new binary async. protocol implementation.

These functions may be re-implemented later in a much simpler manner.

### CODE REVIEW

File: device.c [hunk 1]
```diff
@@ -650,6 +665,9 @@ ssize_t iio_device_debug_attr_read(const struct iio_device *dev,
 ssize_t iio_device_debug_attr_write_raw(const struct iio_device *dev,
 		const char *attr, const void *src, size_t len)
 {
+	if (!attr)
+		return -EINVAL;
```
Comment (nunojsa) [inline device.c:669]: nit: It looks like these little additions have not much to do with this specific patch? Maybe it would make sense to be in a separate patch. They are simple enough though, so fine by me...

**Review by nunojsa — APPROVED** [review]

File: device.c (hunk 1 above)
Comment (pcercuei) [reply device.c:669]: @nunojsa they actually do have to see with the patch - all the attribute read/write did support bulk read/write of all attributes, which was activated when attr == NULL, and return everything into one giant buffer.

**Comment by pcercuei** [comment]
CI fails to build but it seems to be since the last merge of the master branch and unrelated to this PR, so I'll merge it now.
